File: testapp/docs/main_array_sum.cpp
#include <stdio.h>
#include <stdlib.h>

extern int array_sum(int []);

template <class T>

void check(T expect,T result)
{
    if (expect == result)
    {
	//printf("Correct:\n Expected %d got %d \n",expect,result);
    }
    else 
    {
	printf("Incorrect:\n Expected %d got %d \n",expect,result);
	exit (0);
   }
}

int main(void)
{
	int result;
        int a[55555] = {1,2,3,0,0};
	result = array_sum(a);
	check(6, result);
	int b[] = {1,2,3,4,5};
	result = array_sum(b);
	check(15,result);
	printf("All Correct\n");
}

File: testapp/java_files/main_array_sum.java
class main_array_sum
{
    public static <E> void check(E expect, E result)
    {
        if(result.equals(expect))
        {
            System.out.println("Output expected "+expect+" and got "+result);
        }
	else
	{
	    System.out.println("Output expected "+expect+" but got "+result);
	    System.exit(1);
	}
    }
    public static void main(String arg[])
    {
       int result;
       Test t = new Test();
       int x[] = {0,0,0,0,0};
       result = t.array_sum(x);
       check(0, result);
       int a[] = {1,2,3,4,5};
       result = t.array_sum(a);
       check(15, result);
       int b[] = {1,2,3,0,0};
       result = t.array_sum(b);
       check(6, result);
       int c[] = {1,1,1,1,1};
       result = t.array_sum(c);
       check(5, result);
    }
}
